src/xpk/core/docker_image.py

```diff
@@ -161,50 +222,43 @@ def setup_docker_image(args) -> tuple[int, str]:
       0 if successful and 1 otherwise.
       Name of the docker image to use.
   """
-  use_base_docker_image = use_base_docker_image_or_docker_image(args)
-
-  docker_image = args.base_docker_image
-  if use_base_docker_image:
-    validate_docker_image_code = validate_docker_image(docker_image, args)
-    if validate_docker_image_code != 0:
-      xpk_exit(validate_docker_image_code)
-    build_docker_image_code, docker_image = build_docker_image_from_base_image(
-        args
-    )
-    if build_docker_image_code != 0:
-      xpk_exit(build_docker_image_code)
-  else:
-    docker_image = args.docker_image
-    validate_docker_image_code = validate_docker_image(args.docker_image, args)
-    if validate_docker_image_code != 0:
-      xpk_exit(validate_docker_image_code)
 
-  return 0, docker_image
+  docker_image = args.docker_image
 
+  if not docker_image and args.base_docker_image:
+    docker_image = args.base_docker_image  # fallback for legacy users
 
-def use_base_docker_image_or_docker_image(args) -> bool:
-  """Checks for correct docker image arguments.
+  if not docker_image:
+    xpk_print(
+        f'No docker image specified, using default: {DEFAULT_DOCKER_IMAGE}'
+    )
+    docker_image = DEFAULT_DOCKER_IMAGE
```

Comment:
I'm wondering if this would still work correctly for pathways headless mode where we do not attach a docker image to the head. Would you be able to test it with that? This recipe should give you a good guide - https://github.com/AI-Hypercomputer/maxtext/blob/main/benchmarks/recipes/pw_headless_mode.py

Reply:
Ok cool, as long as it comes up correctly that sounds good to me.